### PR TITLE
Generalize f64_to_i64

### PR DESCRIPTION
**Pull request by umanwizard**
`f64_to_i64` might be useful in the other direction, and for u64.

### Motivation

  * This PR adds a feature that has not yet been specified.

    generalize `f64_to_i64` to a `TryCastFrom` trait, which is implemented for the following pairs of types:
    * `f64` and `i64`
    * `i64` and `f64`
    * `f64` and `u64`
    * `u64` and `f64`

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - None

### CODE REVIEW

File: src/ore/src/cast.rs
Comment (jkosh44) [inline src/ore/src/cast.rs:199]: Do you need the `paste` macro here?

**Review by jkosh44 — APPROVED** [review]
Very cool

**Review by benesch — COMMENTED** [review]
Amazing, ty! Didn’t look in detail but don’t wait for my approval.

File: src/ore/src/cast.rs
Comment (umanwizard) [reply src/ore/src/cast.rs:199]: Nope, good catch

**Pull request merged**
